### PR TITLE
8332402: [IR Framework] Add tests for applyIfCPUFeature* and applyIfPlatform* in TestBadFormat

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you help to review this patch to add some test for IR test framework?
As discussed https://github.com/openjdk/jdk/pull/19270#pullrequestreview-2060974799, it's worth to add some tests for for applyIfCPUFeature* and applyIfPlatform* in TestBadFormat.

Thanks

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8332402](https://bugs.openjdk.org/browse/JDK-8332402): [IR Framework] Add tests for applyIfCPUFeature* and applyIfPlatform* in TestBadFormat (**Enhancement** - P4)

### Reviewers
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19302/head:pull/19302` \
`$ git checkout pull/19302`

Update a local copy of the PR: \
`$ git checkout pull/19302` \
`$ git pull https://git.openjdk.org/jdk.git pull/19302/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19302`

View PR using the GUI difftool: \
`$ git pr show -t 19302`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19302.diff">https://git.openjdk.org/jdk/pull/19302.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19302#issuecomment-2132967233)

### CODE REVIEW

**Review by luhenry — APPROVED** [review]

**Comment by luhenry** [comment]
@chhagedorn @eme64 could you please review as you did the most recent changes on this file? Not sure who else to ping on this. Thank you!

File: test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestBadFormat.java [hunk 1]
```diff
@@ -843,12 +843,330 @@ public void mustSpecifyAtLeastOneConstraint2() {
...
+    @IR(failOn = IRNode.CALL,
+        applyIfCPUFeatureAnd = {"sve", "true", "avx", "true"},
+        applyIfCPUFeatureOr = {"sve", "true", "avx", "true"})
+    @IR(failOn = IRNode.CALL,
+        applyIfCPUFeature = {"sve", "true"},
+        applyIfCPUFeatureAnd = {"sve", "true", "avx", "true"},
+        applyIfCPUFeatureOr = {"sve", "true", "avx", "true"})
+    public void onlyOneApplyIfCPUFeature() {}
+
+    @FailCount(3)
```
Comment (eme64) [inline test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestBadFormat.java:861]: Why are there 3 failures?

Comment (eme64) [reply test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestBadFormat.java:861]: Generally, the count is not always as I would expect. Can we fix that? You should at least comment what fails, and why.

I would expect there to be a single failure per rule. But that seems not to be the case.

File: test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestBadFormat.java [hunk 2]
```diff
@@ -843,12 +843,330 @@ public void mustSpecifyAtLeastOneConstraint2() {
...
+    @IR(failOn = IRNode.CALL, applyIfCPUFeatureOr = {"sve", "<0", "avx", "true"})
+    @IR(failOn = IRNode.CALL, applyIfCPUFeatureOr = {"sve", ">=0", "avx", "true"})
+    @IR(failOn = IRNode.CALL, applyIfCPUFeatureOr = {"sve", "<=0", "avx", "true"})
+    public void applyIfCPUFeatureOrFaultyComparator() {}
+
+
     @FailCount(3)
     @Test
-    @IR(failOn = IRNode.CALL, applyIf = {"TLABRefillWasteFraction", "50"}, applyIfNot = {"UseTLAB", "true"})
-    @IR(failOn = IRNode.CALL, applyIfAnd = {"TLABRefillWasteFraction", "50", "UseTLAB", "true"},
```
Comment (eme64) [inline test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestBadFormat.java:849]: Not sure why you changed the format here. You seem to have one-liners everywhere else? Plus it just creates noise when reviewing.

**Review by eme64 — CHANGES_REQUESTED** [review]
Thanks for the work! Generally looks good. I have a few comments.
We should also wait a week, @chhagedorn is out of the office. He generally has more context on the IR framework.

**Review by chhagedorn — COMMENTED** [review]
Thanks for adding all these new tests! I have some comments.

**Pull request closed without merging**